code/Test_definitions/Location-Retrieval.feature

```diff
@@ -0,0 +1,264 @@
+    And the response property "$.message" contains a user friendly text
+
+  @location_retrieval_14_device_identifiers_mismatch
+  Scenario: Device identifiers mismatch
+    # To test this, at least 2 types of identifiers have to be provided, e.g. a phoneNumber and the IP address of a device associated to a different phoneNumber
+    Given that config_var "identifier_types_unsupported" contains at least 2 items
+    And the request body property "$.device" includes several identifiers, each of them identifying a valid but different device
+    When the HTTP "POST" request is sent
+    Then the response status code is 422
+    And the response property "$.status" is 422
+    And the response property "$.code" is "DEVICE_IDENTIFIERS_MISMATCH"
+    And the response property "$.message" contains a user friendly text
+
+  @location_retrieval_15_device_token_mismatch
+  Scenario: Inconsistent access token context for the device
+    # To test this, a token have to be obtained for a different device
+    Given the request body property "$.device" is set to a valid testing device
+    And the header "Authorization" is set to a valid access token emitted for a different device
+    When the HTTP "POST" request is sent
+    Then the response status code is 403
+    And the response property "$.status" is 403
+    And the response property "$.code" is "INVALID_TOKEN_CONTEXT"
+    And the response property "$.message" contains a user friendly text
+
+  @location_retrieval_16_device_not_supported
+  Scenario: Service not available for the device
+    Given that service is not supported for all devices commercialized by the operator 
+    And the request body property "$.device" is set to a valid device for which the service is not applicable
+    When the HTTP "POST" request is sent
+    Then the response status code is 422
+    And the response property "$.status" is 422
+    And the response property "$.code" is "DEVICE_NOT_APPLICABLE"
+    And the response property "$.message" contains a user friendly text
+
+  # Generic 400 errors
+
+  @location_retrieval_400.1_no_request_body
+  Scenario: Missing request body
+    Given the request body is not included
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @location_retrieval_400.2_empty_request_body
+  Scenario: Empty object as request body
+    Given the request body is set to "{}"
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  # Other specific 400 errors
+
+  @location_retrieval_400.3_max_age_schema_compliant

```

Comment:
What do you think about one more test case to validate the maxAge minimum value compliance (minimum=60)?

It could either be combined with this test case (@location_retrieval_400.3_max_age_schema_compliant) or it could be a new test case. Examples below.

Combining into one:
```
@location_retrieval_400.3_max_age_schema_compliant
Scenario: maxAge value does not comply with the schema
    Given the request body property "$.device" is set to a valid testing device
    And the request body property "maxAge" does not comply with the OAS schema at "/components/schemas/RetrievalLocationRequest" 
    When the HTTP "POST" request is sent
    Then the response status code is 400
    And the response property "$.status" is 400
    And the response property "$.code" is "INVALID_ARGUMENT"
    And the response property "$.message" contains a user friendly text

    Examples:
      | maxAge  |
      | 6a0     |
      | 59      |
```

Creating a separate use case:

```
@location_retrieval_400.4_max_age_min_schema_compliant
Scenario: maxAge value is below the minimum (60) allowed
    Given the request body property "$.device" is set to a valid testing device
    And the "maxAge" is set to 59
    When the HTTP "POST" request is sent
    Then the response status code is 400
    And the response property "$.status" is 400
    And the response property "$.code" is "INVALID_ARGUMENT"
    And the response property "$.message" contains a user friendly text

```

Reply:
@trehman-gsma we have removed the maxAge limitation. You can omit it or pass it with 0 value as well.

Reply:
oh! I must have been looking at an older spec then, apologies!